src/ndarray/ndarray.cc

```diff
@@ -1663,7 +1663,7 @@ bool NDArray::LegacyLoad(dmlc::Stream *strm, const uint32_t magic) {
   // load shape
   mxnet::TShape shape;
   if (!LegacyTShapeLoad(strm, &shape, magic)) return false;
-  if (shape.ndim() == 0) {
+  if (mxnet::op::shape_is_none(shape)) {
```

Comment:
@reminisce Not sure if it is the correct fix. It is in save/load.

Reply:
This should be fine.